src/spec-common/secrets.ts

```diff
@@ -0,0 +1,45 @@
+import { Log, LogLevel } from '../spec-utils/log';
+import { CLIHost } from './commonUtils';
+import * as jsonc from 'jsonc-parser';
+import { ContainerError } from './errors';
+import { ParseError } from 'jsonc-parser';
+
+const MASK = '********';
+
+export async function readSecretsFromFile(params: { output?: Log; secretsFile?: string; cliHost: CLIHost }) {
+	const { secretsFile, cliHost, output } = params;
+	if (!secretsFile) {
+		return {};
+	}
+
+	try {
+		const fileBuff = await cliHost.readFile(secretsFile);
+		const parseErrors: ParseError[] = [];
+		const secrets = jsonc.parse(fileBuff.toString(), parseErrors) as Record<string, string>;
+		if (parseErrors.length) {
+			throw new Error('Invalid json data');
+		}
+
+		return secrets;
+	}
+	catch (e) {
+		if (output) {
+			output.write(`Failed to read/parse secrets from file '${secretsFile}'`, LogLevel.Error);
+		}
+
+		throw new ContainerError({
+			description: 'Failed to read/parse secrets',
+			originalError: e
+		});
+	}
+}
+
+export async function maskSecrets(secretsP: Promise<Record<string, string>>, text: string) {
+	const secretValues = Object.values(await secretsP);
+	secretValues.forEach(secret => {
+		const regex = new RegExp(secret, 'g');
```

Comment:
`secret` needs some escaping. Maybe extract a helper function for common use from: https://github.com/devcontainers/cli/blob/236b85162a945a1af9e72fcbe02eb5c7c864b31d/src/spec-shutdown/dockerUtils.ts#L349.

Reply:
What does this escaping actually do?

Reply:
The RexExp will interpret its input string as a regular expression, the escaping takes care of the regex special characters by prepending them with a backslash.

Reply:
That makes sense. I will address this suggestion in a new commit int he morning